rasa/utils/tensorflow/model_data.py

```diff
@@ -323,14 +361,18 @@ def split(
             # this operation can be performed only for labels
             # that contain several data points
             multi_values = [
-                f[counts > 1]
+                list(np.array(f)[counts > 1])
+                if RasaModelData.is_in_4d_format(f)
+                else f[counts > 1]
                 for attribute_data in self.data.values()
                 for features in attribute_data.values()
                 for f in features
             ]
             # collect data points that are unique for their label
             solo_values = [
-                f[counts == 1]
+                list(np.array(f)[counts == 1])
```

Comment:
that is a very strange thing 😂, I'd say it'd be better to use some special structure, to avoid things like that